products/BellHybrid/services/desktop/endpoints/deviceInfo/DeviceInfoEndpoint.cpp

```diff
@@ -14,15 +14,27 @@
 #include <vector>
 #include <sys/statvfs.h>
 #include <purefs/filesystem_paths.hpp>
+#include <purefs/vfs_subsystem.hpp>
 
 #include <ctime>
+namespace
+{
+    constexpr auto DEVICE_NAME    = "emmc0sys1";
+    constexpr auto EMMC_SN_LENGTH = 13;
 
+    inline void readSerialNumberFromEmmc(char *serial_number)
+    {
+        purefs::subsystem::disk_mgr()->read(DEVICE_NAME, serial_number, 2, 1U);
```

Comment:
I'm almost sure this function causes memory corruption - `read` gets the entire block from eMMC card, which AFAIR has 512 bytes, but here you try to store it in 13-bytes array.

Reply:
Okay, I will change it to 512 and check.

Reply:
Done